src/commands/scan/handle-create-new-scan.mts

```diff
@@ -153,3 +179,109 @@ export async function handleCreateNewScan({
     await outputCreateNewScan(fullScanCResult, outputKind, interactive)
   }
 }
+
+async function performReachabilityAnalysis({
+  branchName,
+  cwd,
+  interactive,
+  orgSlug,
+  outputKind,
+  packagePaths,
+  repoName,
+}: {
+  packagePaths: string[]
+  orgSlug: string
+  cwd: string
+  repoName: string
+  branchName: string
+  outputKind: OutputKind
+  interactive: boolean
+}): Promise<{ ok: boolean; scanPaths?: string[] }> {
+  logger.info('Starting reachability analysis...')
+
+  packagePaths = packagePaths.filter(
+    p =>
+      /* Exclude DOT_SOCKET_DOT_FACTS_JSON from previous runs */ !p.includes(
+        constants.DOT_SOCKET_DOT_FACTS_JSON,
+      ),
+  )
+
+  // Lazily access constants.spinner.
+  const { spinner } = constants
+
+  // Setup SDK for uploading manifests
+  const sockSdkCResult = await setupSdk()
+  if (!sockSdkCResult.ok) {
+    await outputCreateNewScan(sockSdkCResult, outputKind, interactive)
+    return { ok: false }
```

Comment:
So how this flow generally ought to work is that you return a CResult here, in this case something like `{ok:false, message: 'reason'}` and then the handler would pass on that result to the output function which will see the ok:false and emit the warning. You're kind of taking a shortcut by calling it directly but should be unnecessary.